R/derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,528 @@
+#'
+#'       - `"CR"`: An assessment is considered as complete response (CR) if
+#'           - `AVALC == "CR"`,
+#'           - there is a confirmatory assessment with `AVALC == "CR"` at least
+#'           `ref_confirm` days after the assessment,
+#'           - all assessments between the assessment and the confirmatory
+#'           assessment are `"CR"` or `"NE"`, and
+#'           - there are at most `max_nr_ne` `"NE"` assessments between the
+#'           assessment and the confirmatory assessment.
+#'
+#'       - `"PR"`: An assessment is considered as partial response (PR) if
+#'           - `AVALC == "PR"`,
+#'           - there is a confirmatory assessment with `AVALC %in% c("CR",
+#'           "PR")` at least `ref_confirm` days after the assessment,
+#'           - all assessments between the assessment and the confirmatory
+#'           assessment are `"CR"`, `"PR"`, or `"NE"`,
+#'           - there is no `"PR"` assessment after a `"CR"` assessment in the
+#'           confirmation period, and
+#'           - there are at most `max_nr_ne` `"NE"` assessments between the
+#'           assessment and the confirmatory assessment.
+#'
+#'           If the `accept_sd` argument is set to `TRUE`, one `"SD"` assessment
+#'           in the confirmation period is accepted.
+#'
+#'        - `"SD"`: An assessment is considered as stable disease (SD) if
+#'          - `AVALC %in% c("CR", "PR", "SD")` and
+#'          - the assessment is at least `ref_start_window` days after
+#'          `reference_date`.
+#'
+#'        - `"NON-CR/NON-PD"`: An assessment is considered as NON-CR/NON-PD if
+#'          - `AVALC = "NON-CR/NON-PD"` and
+#'          - the assessment is at least `ref_start_window` days after
+#'          `reference_date`.
+#'
+#'        - `"PD"`: An assessment is considered as progressive disease (PD) if
+#'        `AVALC == "PD"`.
+#'
+#'        - `"NE"`: An assessment is considered as not estimable (NE) if
+#'            - `AVALC == "NE"` or
+#'            - `AVALC %in% c("CR", "PR", "SD")` and the assessment is less than
+#'            `ref_start_window` days after `reference_date`.
+#'
+#'        - `"ND"`: An assessment is considered as not done (ND) if `AVALC ==
+#'        "ND"`.
+#'
+#'        - `"MISSING"`: An assessment is considered as missing (MISSING) if a
+#'        subject has no observation in the input dataset.
+#'
+#'            If the `missing_as_ne` argument is set to `TRUE`, `AVALC` is set to
+#'            `"NE"` for missing assessments.
+#'
+#'   1. For each subject the best response as derived in the previous step is
+#'   selected, where `"CR"` is best and `"MISSING"` is worst in the order above.
+#'   If the best response is not unique, the first one (with respect to `ADT`)
+#'   is selected. Only the variables `AVALC`, `ADT`, and the variables specified
+#'   for `subject_keys` are kept from the selected observations.
+#'
+#'   1. The `AVAL` variable is added and set to `aval_fun(AVALC)`.
```

Comment:
i think this step can be removed as AVAL should already have been derived for RS records so could just keep this in the newly created records

Reply:
We need to set it because it may differ from the value of the source record. For example if BOR is SD and the source record was PR (unconfirmed), the `AVAL` value for SD should be used and not the `AVAL` value for PR.

Reply:
i see. i'm more just questioning does it need to be an argument with user flexibility, or could we just say in our documentation that AVAL is set as: 
_Set to 1 if [ADRS.AVALC]='CR'
Else set to 2 if [ADRS.AVALC]='PR'
Else set to 3 if [ADRS.AVALC]='SD'
Else set to 4 if [ADRS.AVALC]='NON-CR/NON-PD'
Else set to 5 if [ADRS.AVALC]='PD'
Else set to 6 if [ADRS.AVALC]='NE'_

Reply:
See my comment above.